src/Microsoft.AspNet.Mvc.Core/ModelBinding/ArrayModelBinder.cs

```diff
@@ -25,15 +27,29 @@ public override Task<ModelBindingResult> BindModelAsync([NotNull] ModelBindingCo
             return base.BindModelAsync(bindingContext);
         }
 
-        protected override object CreateEmptyCollection()
+        /// <inheritdoc />
+        public override bool CanCreateInstance(Type targetType)
+        {
+            Debug.Assert(targetType == typeof(TElement[]), "GenericModelBinder only creates this binder for arrays.");
+
+            return true;
```

Comment:
Why just do the thing you put in the assert? You could say with like 20 characters what all of this does, and technically be more correct :laughing: